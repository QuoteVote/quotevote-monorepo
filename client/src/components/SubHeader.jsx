import PropTypes from 'prop-types'
import Tooltip from '@mui/material/Tooltip'
import { styled } from '@mui/material/styles'
import { Grid, Typography } from '@mui/material'

const StyledTooltip = styled(({ className, ...props }) => (
  <Tooltip {...props} classes={{ popper: className }} />
))(() => ({
  '& .MuiTooltip-tooltip': {
    backgroundColor: '#20e08e',
  },
}))

const ValueLabelComponent = (props) => {
  const { children, open, value } = props
  const prefix = value < 0 ? '-' : '+'
  return (
    <StyledTooltip
      open={open}
      enterTouchDelay={0}
      placement="top"
      title={`${prefix} ${value}`}
      arrow
    >
      {children}
    </StyledTooltip>
  )
}

ValueLabelComponent.propTypes = {
  children: PropTypes.element.isRequired,
  open: PropTypes.bool.isRequired,
  value: PropTypes.number.isRequired,
}

export default function SubHeader({
  headerName,
  showFilterIconButton = true,
  setOffset,
}) {
  return (
    <Grid
      container
      direction="row"
      justify="center"
      alignItems="center"
      sx={{ mt: 1, height: '85px', borderRadius: '6px' }}
    >
      <Grid item xs={12} sm={3} md={3}>
        <Typography
          sx={{
            color: '#424556',
            fontFamily: 'Montserrat',
            fontWeight: 'bold',
            height: '28px',
            fontSize: { xs: '18px', sm: '24px' },
            textAlign: { xs: 'left', sm: 'center' },
            pl: { xs: 0.625, sm: 0 },
          }}
        >
          {headerName}
        </Typography>
      </Grid>
      <Grid item xs={8} sm={5} md={6} />
      <Grid item xs={4} sm={3} md={3} />
    </Grid>
  )
}

SubHeader.propTypes = {
  headerName: PropTypes.string.isRequired,
  showFilterIconButton: PropTypes.bool,
  setOffset: PropTypes.any,
}
