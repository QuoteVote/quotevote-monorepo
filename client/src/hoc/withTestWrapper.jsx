import React from 'react'
import { PersistGate } from 'redux-persist/integration/react'
import { ApolloProvider } from '@apollo/react-hooks'
import { Provider } from 'react-redux'
import { createTheme, ThemeProvider } from '@material-ui/core/styles'
import { Router } from 'react-router-dom'
import { createBrowserHistory } from 'history'
import store, { persistor } from '../store/store'
import client from '../config/apollo'
import customTheme from '../theme'
import { AuthModalProvider } from '../Context/AuthModalContext'
const hist = createBrowserHistory()
const theme = createTheme(customTheme)

const withTestWrapper = (Component) => (props) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <ThemeProvider theme={theme}>
          <AuthModalProvider>
            <Router history={hist}>
              <Component {...props} />
            </Router>
          </AuthModalProvider>
        </ThemeProvider>
      </PersistGate>
    </Provider>
  </ApolloProvider>
)
export default withTestWrapper
