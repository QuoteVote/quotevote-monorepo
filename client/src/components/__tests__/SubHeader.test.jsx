/* eslint-disable no-undef */
/* eslint-disable react/jsx-no-undef */
import React from 'react'
import { render } from '@testing-library/react'
import SubHeader from '../SubHeader'
import withTestWrapper from '../../hoc/withTestWrapper'

const SubHeaderWrapper = withTestWrapper((props) => <SubHeader headerName="" {...props} />)

describe('SubHeader test -', () => {
  it('renders correctly', async () => {
    let container
    await act(async () => {
      const res = render(
        <SubHeaderWrapper />,
      )
      container = res.container
    })
    expect(container.firstChild).toMatchSnapshot()
  })
})
