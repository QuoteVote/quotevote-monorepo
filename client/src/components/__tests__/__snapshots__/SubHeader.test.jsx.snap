// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubHeader test - > renders correctly 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row makeStyles-header-1 css-wjsns7-MuiGrid-root"
  justify="center"
>
  <div
    class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 makeStyles-headerName-2 css-rizt0-MuiTypography-root"
    />
  </div>
  <div
    class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
  />
  <div
    class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root"
  />
</div>
`;
